test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/UseOtlpExporterExtensionTests.cs

```diff
@@ -0,0 +1,345 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#nullable enable
+
+using System.Diagnostics;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Options;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Tests;
+using OpenTelemetry.Trace;
+using Xunit;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests;
+
+public class UseOtlpExporterExtensionTests
+{
+    [Fact]
+    public void UseOtlpExporterDefaultTest()
+    {
+        var services = new ServiceCollection();
+
+        services.AddOpenTelemetry()
+            .UseOtlpExporter();
+
+        using var sp = services.BuildServiceProvider();
+
+        var exporterOptions = sp.GetRequiredService<IOptionsMonitor<OtlpExporterBuilderOptions>>().CurrentValue;
+
+        Assert.Equal(new Uri(OtlpExporterOptions.DefaultGrpcEndpoint), exporterOptions.DefaultOptions.Endpoint);
+        Assert.Equal(OtlpExporterOptions.DefaultOtlpExportProtocol, exporterOptions.DefaultOptions.Protocol);
+        Assert.False(((OtlpExporterOptions)exporterOptions.DefaultOptions).HasData);
+
+        Assert.False(((OtlpExporterOptions)exporterOptions.LoggingOptions).HasData);
+        Assert.False(((OtlpExporterOptions)exporterOptions.MetricsOptions).HasData);
+        Assert.False(((OtlpExporterOptions)exporterOptions.TracingOptions).HasData);
+    }
+
+    [Fact]
+    public void UseOtlpExporterSetProtocolTest()
+    {
+        var services = new ServiceCollection();
+
+        services.AddOpenTelemetry()
+            .UseOtlpExporter(OtlpExportProtocol.HttpProtobuf);
+
+        using var sp = services.BuildServiceProvider();
+
+        var exporterOptions = sp.GetRequiredService<IOptionsMonitor<OtlpExporterBuilderOptions>>().CurrentValue;
+
+        Assert.Equal(OtlpExportProtocol.HttpProtobuf, exporterOptions.DefaultOptions.Protocol);
+        Assert.Equal(new(OtlpExporterOptions.DefaultHttpEndpoint), exporterOptions.DefaultOptions.Endpoint);
+        Assert.True(((OtlpExporterOptions)exporterOptions.DefaultOptions).HasData);
+
+        Assert.False(((OtlpExporterOptions)exporterOptions.LoggingOptions).HasData);
+        Assert.False(((OtlpExporterOptions)exporterOptions.MetricsOptions).HasData);
+        Assert.False(((OtlpExporterOptions)exporterOptions.TracingOptions).HasData);
+    }
+
+    [Fact]
+    public void UseOtlpExporterSetEndpointAndProtocolTest()
```

Comment:
Mark this test and `UseOtlpExporterSetProtocolTest` as `Theory` to test both the protocols.

Reply:
I went ahead and made this change but testing `OtlpExportProtocol.Grpc` is a bit tricky because that is the default value. The test really can't know for certain if the protocol was applied or just the default was used. With `OtlpExportProtocol.HttpProtobuf` it knows for sure.

Reply:
> The test really can't know for certain if the protocol was applied or just the default was used.

I think that's fine. This test should only be concerned about whether or not the exporter uses gRPC when the user calls `UseOtlpExporter` with the gRPC as the value for protocol parameter. We need not worry about whether it used the default value or rewrite the same value for this test.